spring-cloud-gcp-security-iap/src/main/java/org/springframework/cloud/gcp/security/iap/AppEngineAudienceSupplier.java

```diff
@@ -30,19 +32,33 @@
  *
  * @since 1.1
  */
-public class AppEngineAudienceValidator extends AudienceValidator {
+public class AppEngineAudienceSupplier implements Supplier<String> {
```

Comment:
This is not good. There is a Spring Cloud Function project (https://cloud.spring.io/spring-cloud-function) which detects all the `java.function` interfaces and do some stuff around them. Therefore this one being auto-configured is going to be at least a noise in the Function Registry. I don't tell already that there might be some endpoint exposure or some other unexpected behavior.

Reply:
The suppliers are not exposed as autoconfigured beans; they are directly instantiated and passed to AudienceValidator objects (which are exposed as beans in autoconfiguration) in `IapAuthenticationAutoConfiguration`.

I guess I could create a special-purpose interface for AudienceProvider instead of implementing Supplier<String> directly. Would that help?